### DISCUSSION TITLE
Continuous Aggregates - User feedback

### DISCUSSION THREAD

**Discussion by ramonguiu** [timescale/promscale#828] [Feedback on Features]
Promscale continuous aggregates leverage a TimescaleDB feature of the same name to have the database manage downsampling and data materialization.

Read more details in the [launch blog post](https://blog.timescale.com/blog/a-different-and-often-better-way-to-downsample-your-prometheus-metrics/) and [the documentation](https://github.com/timescale/promscale/blob/master/docs/downsampling.md).

The feature is currently in beta and we are using this discussion to capture user feedback. 

You are also welcome to join our [Slack community](https://timescaledb.slack.com/) and post questions in the #promscale channel.

**Comment by toymachiner62** [comment]
I've tried to follow the blog post and i'm hitting road blocks left and right. The functions don't work unless you prefix with `toolkit_experimental` and then if I do that I get an error that `aggregates cannot use named arguments`. 

Here's a link to my issue in slack https://timescaledb.slack.com/archives/C4GT3N90X/p1635305238080800

**Reply by ramonguiu** [reply]
Thanks for the feedback @toymachiner62 . What version of the toolkit are you using? You can use this query: `SELECT extname, extversion FROM pg_extension;`
I think you may be using an old one. 

I can resolve the error if I remove `bounds =>` from the query. I'm checking internally why we are using it and I'll provide an update.

**Reply by toymachiner62** [reply]
```
plpgsql	1.0
promscale	0.1.2
timescaledb	2.4.2
timescaledb_toolkit	1.1
```

We're using managed timescaledb (as opposed to timescale cloud I think)

**Comment by toymachiner62** [comment]
This view creation does not work as documented in the blog post. It's missing the `AS`

```
CREATE OR REPLACE VIEW cpu_usage_30m 
SELECT
    bucket + '30 min' as time,
    series_id,
    extrapolated_delta(cnt_agg, method =>'prometheus') as increase,
    extrapolated_rate(cnt_agg, method => 'prometheus') as rate,
    irate_right(cnt_agg) as irate,
    num_resets(cnt_agg)::float as resets
FROM cpu_usage_30m_aggregate;
```

**Reply by toymachiner62** [reply]
Also after creating that view the query doesn't work as described in the blog post

![image](https://user-images.githubusercontent.com/485782/139258421-2d716d60-b719-43b0-afb3-44ae03413384.png)

**Comment by ramonguiu** [comment]
@toymachiner62 sorry for the delay and thanks so much for your feedback. We have updated the blog post and you should be able to run those queries now.

**Comment by mwolter805** [comment]
Hello, How would you recommend aggregating a Prometheus metric that only has a binary state? On and off, or actually 0 and 1.

Thanks!
